cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/store/ActionCreator.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import { GLOBALS } from 'services/global-constants';
+import { MyPipelineApi } from 'api/pipeline';
+import Store, { Actions } from 'components/PipelineList/DeployedPipelineView/store';
+import { objectQuery } from 'services/helpers';
+import { PROGRAM_STATUSES } from 'services/global-constants';
+import {
+  IPipeline,
+  IPipelineStatus,
+  IStatusMap,
+  IRunsCountMap,
+} from 'components/PipelineList/DeployedPipelineView/types';
+
+enum ProgramType {
+  'cdap-data-pipeline' = 'Workflow',
+  'cdap-data-streams' = 'Spark',
+}
+
+interface IPipelineParams {
+  appId: string;
+  programType: string;
+  programId: string;
+}
+
+const DEFAULT_STATUS: IPipelineStatus = {
+  status: PROGRAM_STATUSES.DEPLOYED,
+  lastStarting: null,
+};
+
+export function fetchPipelineList() {
+  const namespace = getCurrentNamespace();
+
+  const params = {
+    namespace,
+    artifactName: GLOBALS.etlPipelineTypes.join(','),
+  };
+
+  MyPipelineApi.list(params).subscribe((res: IPipeline[]) => {
+    Store.dispatch({
+      type: Actions.setPipeline,
+      payload: {
+        pipelines: res,
+      },
+    });
+
+    fetchRunsInfo();
+  });
+}
+
+export function deletePipeline(pipeline: IPipeline) {
+  const namespace = getCurrentNamespace();
+
+  const params = {
+    namespace,
+    appId: pipeline.name,
+  };
+
+  MyPipelineApi.delete(params).subscribe(fetchPipelineList);
+}
+
+function fetchRunsInfo() {
+  const namespace = getCurrentNamespace();
+  const pipelines: IPipelineParams[] = Store.getState().deployed.pipelines.map((pipeline) => {
+    const programInfo = GLOBALS.programInfo[pipeline.artifact.name];
+
+    return {
+      appId: pipeline.name,
+      programType: ProgramType[pipeline.artifact.name],
+      programId: programInfo.programName,
+    };
+  });
+
+  fetchRuns(namespace, pipelines);
+  fetchRunsCount(namespace, pipelines);
+}
+
+/**
+ *
+ * @param namespace
```

Comment:
👍 We should start doing this. TS-vscode integration actually shows these comments while inspecting the function in its usage.